src/util/c_typecast.cpp

```diff
@@ -616,8 +616,8 @@ void c_typecastt::implicit_typecast_followed(
     // to prevent anything that may surprise us.
 
     if (
-      ns.follow(dest_type.subtype()).id() ==
-      src_type.id()) // TODO: remove this condition
+      ns.follow(dest_type.subtype()).id() == src_type.id() &&
+      src_type != dest_type) // TODO: remove this condition
```

Comment:
This is weird to me. If `src_type == dest_type`, (a) why would this even be executed, and (b) what would *not* executing it change? I'm confused, sorry.